services/src/main/java/org/keycloak/device/DeviceRepresentationProviderFactoryImpl.java

```diff
@@ -14,17 +31,37 @@ public String getId() {
         return PROVIDER_ID;
     }
 
+    @SuppressWarnings("NullableProblems")
+    @Override
+    public void init(Config.Scope config) {
+        Caffeine<Object, Object> builder = Caffeine.newBuilder()
+                .maximumSize(config.getInt(CACHE_SIZE, DEFAULT_CACHE_SIZE))
+                .softValues();
+
+        if (config.root().getBoolean(MetricsOptions.METRICS_ENABLED.getKey(), Boolean.FALSE)) {
+            var stats = new CaffeineStatsCounter(Metrics.globalRegistry, "user-agent");
```

Comment:
Should we make the metric name `keycloak.user-agent` or maybe even `keycloak.device.user-agent` to better match the existing metric conventions https://www.keycloak.org/observability/metrics-for-troubleshooting-keycloak

Reply:
Only if I won't use `CaffeineStatsCounter` and implement it myself. 
Quarkus metrics don't start with `keycloak` either.

Reply:
No at all. It simply demonstrates that the metric isn't useful, as it doesn't matter for troubleshooting or capacity planning. At least, I can't come up with a good story for it :shrug: 
Removing it.